src/UI/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -40,19 +41,22 @@ public class InputModel
         public virtual Task<IActionResult> OnPostConfirmationAsync(string returnUrl = null) => throw new NotImplementedException();
     }
 
-    internal class ExternalLoginModel<TUser> : ExternalLoginModel where TUser : IdentityUser, new()
+    internal class ExternalLoginModel<TUser> : ExternalLoginModel where TUser : class
     {
         private readonly SignInManager<TUser> _signInManager;
         private readonly UserManager<TUser> _userManager;
+        private readonly IUIUserInstanceFactory<TUser> _userInstanceFactory;
```

Comment:
Please, do not introduce a new abstraction. Keep the new() constraint around instead

Reply:
How would a new constraint help when to be able to save a user both a UserName / and or Email must be set as well.

Reply:
Use https://github.com/aspnet/Identity/blob/85f8a49aef68bf9763cd9854ce1dd4a26a7c5d3c/src/Stores/UserStoreBase.cs#L152 instead

Reply:
Nope: https://github.com/aspnet/Identity/blob/85f8a49aef68bf9763cd9854ce1dd4a26a7c5d3c/src/Stores/UserStoreBase.cs#L35

Those are tied to identity user as well.

Reply:
https://github.com/aspnet/Identity/blob/85f8a49aef68bf9763cd9854ce1dd4a26a7c5d3c/src/Core/IUserStore.cs#L39

Reply:
I meant to use them through the interface, just pointed to the implementation so that Kiran could see what they were doing